src/main/java/frc/robot/sim/DrivebaseSim.java

```diff
@@ -1,152 +1,186 @@
-          case HALValue.kUnassigned:
-            System.out.println(device.name + "." + value.name + ": (unassigned)");
-            break;
-          default:
-            System.out.println(device.name + "." + value.name + ": (unknown): " + value.value.getLong());
-        }
-      }
-    }
-  }
-}
+package frc.robot.sim;
+
+import javax.swing.text.Position;
+
+import com.revrobotics.CANSparkMax;
+
+import edu.wpi.first.hal.HALValue;
+import edu.wpi.first.hal.SimDouble;
+import edu.wpi.first.hal.simulation.SimDeviceDataJNI;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.DifferentialDrivetrainSim;
+import frc.robot.Constants.Drive;
+
+/**
+ * Since REVPhysicsSim doesn't work, let's try to implement
+ * our own simulation that updates the relevant simulation
+ * variables for Spark MAX motors.
+ */
+public class DrivebaseSim {
+  // Simulation value handles that are doubles
+  public static final String APPLIED_OUTPUT = "Applied Output";
+  public static final String VELOCITY = "Velocity";
+  public static final String VELOCITY_CONVERSION_FACTOR = "Velocity Conversion Factor";
+  public static final String POSITION = "Position";
+  public static final String POSITION_CONVERSION_FACTOR = "Position Conversion Factor";
+  public static final String BUS_VOLTAGE = "Bus Voltage"; // Always 12.0?
+  public static final String MOTOR_CURRENT = "Motor Current";
+  public static final String ANALOG_VOLTAGE = "Analog Voltage";
+  public static final String ANALOG_VELOCITY = "Analog Velocity";
+  public static final String ANALOG_POSITION = "Analog Position";
+  public static final String ALT_ENCODER_VELOCITY = "Alt Encoder Velocity";
+  public static final String ALT_ENCODER_POSITION = "Alt Encoder Position";
+  public static final String STALL_TORQUE = "Stall Torque";
+  public static final String FREE_SPEED = "Free Speed";
+
+  // Simulation value handles that are ints
+  public static final String FAULTS = "Faults";
+  public static final String STICKY_FAULTS = "Sticky Faults";
+  public static final String MOTOR_TEMPERATURE = "Motor Temperature"; // Always 25?
+  public static final String CONTROL_MODE = "Control Mode";
+  public static final String FW_VERSION = "FW Version";
+
+  private CANSparkMax m_leftMotor;
+  private CANSparkMax m_rightMotor;
+
+  private int m_leftMotorDeviceHandle;
+  private int m_rightMotorDeviceHandle;
+
+  private SimDouble m_leftAppliedOutput;
+  private SimDouble m_rightAppliedOutput;
+
+  private SimDouble m_leftPositionMeters;
+  private SimDouble m_rightPositionMeters;
+
+  private SimDouble m_leftVelocityMetersPerSecond;
+  private SimDouble m_rightVelocityMetersPerSecond;
+
+  // Suggested code from
+  // https://pdocs.kauailabs.com/navx-mxp/software/roborio-libraries/java/
+  public static final String GYRO_DEVICE_NAME = "navX-Sensor[0]";
+  private SimDouble m_yaw;
+
+  private DifferentialDrivetrainSim m_drivetrainSim;
+
+  /**
```

Comment:
There are almost no changes to this file, but it was formatted incorrectly, so that's why so many changes appear.
The relevant changes are moving these two fields here from DrivebaseSubsystem, adding the handling for them in periodic() and inverting the yaw.